web/src/engine/websites/VortexScans.ts

```diff
@@ -66,23 +53,31 @@ export default class extends DecoratableMangaScraper {
     }
 
     public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
-        const data = await FetchWindowScript<MangaID>(new Request(url), mangaIdScript, 4000);
-        return new Manga(this, provider, JSON.stringify({ id: data.id.toString(), slug: data.slug }), data.title);
+        const mangaSlug = url.split('/').pop();
+        const { post: { id, slug, postTitle } } = await FetchJSON<APISingleManga>(new Request(new URL(`post?postSlug=${mangaSlug}`, this.apiUrl)));
+        return new Manga(this, provider, JSON.stringify({ slug, id: id.toString() }), postTitle);
     }
 
     public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
-        const endpoint = new URL('query?perPage=9999', this.apiURL);
-        const { posts } = await FetchJSON<APIMangas>(new Request(endpoint));
-        return posts.map(manga => new Manga(this, provider, JSON.stringify({
-            id: manga.id.toString(),
-            slug: manga.slug
-        }), manga.postTitle.trim()));
+        const mangaList: Manga[] = [];
+        for (let page = 1, run = true; run; page++) {
+            const mangas = await this.GetMangasFromPage(page, provider);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
+    }
+
+    private async GetMangasFromPage(page: number, provider: MangaPlugin): Promise<Manga[]> {
+        const { posts } = await FetchJSON<APIMangas>(new Request(new URL(`query?page=${page}&perPage=200`, this.apiUrl)));
+        return posts.map(item => new Manga(this, provider, JSON.stringify({ slug: item.slug, id: item.id.toString() }), item.postTitle));
     }
 
     public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
-        const mangaid: MangaID = JSON.parse(manga.Identifier);
-        const endpoint = new URL(`chapters/?postId=${mangaid.id}&skip=0&take=99999&order=asc`, this.apiURL);
-        const { post } = await FetchJSON<APIChapters>(new Request(endpoint));
-        return post.chapters.map(chapter => new Chapter(this, manga, `/series/${mangaid.slug}/${chapter.slug}`, [chapter.number.toString(), chapter.title].join(' ').trim()));
+        const { id, slug } = JSON.parse(manga.Identifier) as MangaID;
+        const { post: { chapters } } = await FetchJSON<APISingleManga>(new Request(new URL(`chapters?postId=${id}`, this.apiUrl)));
+        return chapters ? chapters.map(chapter => {
+            const title = chapter.title ? [chapter.number.toString(), ':', chapter.title].join(' ') : chapter.number.toString();
```

Comment:
Usually i would also prefer to use `string.join` to combine a conditional title, but the compact string interpolated variant `` `${chapter.number} : ${chapter.title}` `` seems quite tempting as an alternative

Reply:
Silly me :D